contracts/src/steel/OpSteel.sol

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.9;
+
+import {Encoding, Steel} from "./Steel.sol";
+
+abstract contract OpCommitmentValidator {
+    /// @notice Address of the OptimismPortal2 contract.
+    IOptimismPortal2 public immutable optimismPortal;
+
+    constructor(address _optimismPortal) {
+        optimismPortal = IOptimismPortal2(_optimismPortal);
+    }
+
+    /// @notice Validates if the Steel commitment.
+    /// @return True if the commitment is valid, false otherwise.
+    function validateCommitment(Steel.Commitment memory commitment) internal view returns (bool) {
+        (uint240 blockID, uint16 version) = Encoding.decodeVersionedID(commitment.id);
+        if (version == 0x100) {
+            return validateDisputeGameCommitment(blockID, commitment.digest);
+        } else {
+            return Steel.validateCommitment(commitment);
```

Comment:
One solution to this would be to check the `configID` against an `immutable` value on this contract. We might want to apply this to the functionality in `Steel.sol` as well. I understand that it slightly more difficult because that is currently a library. We might discuss what the right solution is here.

Reply:
I agree 100%, especially with different chains and configID, we should rethink our Steel Solidity design. Unfortunately one library is not enough anymore and we should discuss what the right solution could be.